### ISSUE TITLE
[BUG] Blob Storage URI should not escape segment separator (/)

### ISSUE THREAD

**Issue by mburumaxwell** [Azure/azure-sdk-for-net#18078]
**Describe the bug**
Azure Storage Blob names that are multi-segment (resembles virtual directories) generate absolute URIs that have escapes on the path separator (`/`). If pass the result to another service, the URL may be considered invalid.

**Expected behavior**
The URI escaping should be done as a URI and not a data segment.
Example https://contoso.blob.core.windows.net/backups/db1/20210101.bacpac

**Actual behavior (include Exception or Stack Trace)**
What is the actual behavior?
Example https://contoso.blob.core.windows.net/backups/db1%2F20210101.bacpac

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue).

```cs
var cs = "UseDevelopmentStorage=true;"
var bc = new BlobClient(cs, "azuresqlbackups", "db1/20210101.bacpac");
var bacpacUri = bc.Uri.AbsoluteUri;
```

The [code](https://github.com/Azure/azure-sdk-for-net/blob/6482d50cd4c8101ffbee83bdd5d12457ad6db2c2/sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs#L309) in the `BlobUriBuilder` calls `Uri.EscapeDataString(...)` instead of `Uri.EscapeUriString(...)`.

**Environment:**
 - Name and version of the Library package used: [e.g. Azure.Storage.Blobs 12.8.0] 
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): .NET 5.0.102
 - IDE and version : [e.g. Visual Studio 16.8.4]

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team best able to assist.

**Comment by amnguye** [comment]
Hi, I can reproduce your issue.

@seanmcc-msft What do you think about this? Would it be a breaking change to change the way we are encoding the slashings in the Uri. It is quite odd we are encoding the path but nothing past the container.

**Comment by amnguye** [comment]
So this escaping of the '/' in the blob name in the URI is by design. Please see issue https://github.com/Azure/azure-sdk-for-net/issues/15064 and the other issues that are referenced to it.

**Comment by mburumaxwell** [comment]
Hi @amnguye ,

It makes sense to have the blob name to be escaped. However, unescaping (reverse) can cause other parts that need to remain escaped to be undone. For example: `https://constoso.blob.core.windows.net/backups/db%201%2F202010101.bacpac`
would become `https://constoso.blob.core.windows.net/backups/db 1/202010101.bacpac`.
However, if you use `Uri.EscapeUriString(...)` instead of `Uri.EscapeDataString(...)`, the generated one would be `https://constoso.blob.core.windows.net/backups/db%201/202010101.bacpac`.
Note: it is possible to unescape data string using `Uri.UnescapeDataString(...)` but there is no way to unescape URI strings.

**Comment by amnguye** [comment]
As per the [issue](https://github.com/Azure/azure-sdk-for-net/issues/15064#issuecomment-691749986) I linked, the service considers `db1/20210101.bacpac` to be the same blob as `db1%2F20210101.bacpac`. As the [issue](https://github.com/Azure/azure-sdk-for-net/issues/15064#issuecomment-692131075) goes onto explain, if we did not do this, others regarding encoding blob names correctly and sending the correct URL would not be possible.

We recommend to use `BlobClient.Name` to extract out the name of the blob.
